### DISCUSSION TITLE
How to use AOT compiled shared libraries

### DISCUSSION THREAD

**Discussion by jbarrmetro** [WasmEdge/WasmEdge#2209] [Q&A]
Maybe I'm just dense and missing something obvious, but I read through the whole of the documentation and couldn't find any examples related to using AOT compiled shared libraries. I have a project in mind that WasmEdge is perfect for, but I need the ability to link in libraries either dynamically or during a post-compilation linking phase. Is that something it can do?

**Comment by hydai** [comment]
Hi, change the extension from wasm to shared object (.so).

Then, it will generate the shared object instead of the universal wasm format.

Reference: https://wasmedge.org/book/en/cli/wasmedgec.html#example

**Comment by jbarrmetro** [comment]
I saw that, but is there any way to link in a shared library when running an executable? Or link it in during the compilation phase? It doesn't seem like much use to just have another way to run a standalone file, I would imagine the point of having a shared library is to, erm, share it between programs.

**Reply by hydai** [reply]
Currently, we don't support this feature shortly. We don't release the interface of our built shared object, so it's not easy to use it directly. And we also don't support compiling the wasm into a native executable.
The current design only allows users to execute the shared object with the WasmEdge runtime.
